cmd/shortener/config/flag_test.go

```diff
@@ -1,150 +1,105 @@
+func TestParseFlags(t *testing.T) {
 	tests := []struct {
-		name           string
-		body           string
-		args           map[string]string
-		wantError      string
-		wantStatusCode int
+		name     string
+		args     []string
+		expected Args
 	}{
 		{
-			name: "incorrect base_url:not set scheme into argument shortener_base_url",
-			body: "https://yandex.ru",
-			args: map[string]string{
-				"shortener_server_address": "http://localhost:8080",
-				"shortener_base_url":       "://localhost:8080",
+			name: "parse_flags_correctly",
+			args: []string{
+				"-a=:8081",
+				"-b=http://localhost:8081",
+				"-f=/tmp/storage.txt",
+				"-d=postgres://user:pass@localhost/dbname",
 			},
-			wantError:      "failed to parse base url\n",
-			wantStatusCode: http.StatusBadRequest,
-		},
-		{
-			name: "incorrect base_url:not set port into argument shortener_base_url",
-			body: "https://practicum.yandex.ru",
-			args: map[string]string{
-				"shortener_server_address": "http://localhost:8080",
-				"shortener_base_url":       "http://localhost",
+			expected: Args{
+				ServerAddress:   ":8081",
+				BaseURL:         "http://localhost:8081",
+				FileStoragePath: "/tmp/storage.txt",
+				DatabaseDNS:     "postgres://user:pass@localhost/dbname",
 			},
-			wantError:      "failed to split host and port\n",
-			wantStatusCode: http.StatusBadRequest,
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			args := initArgs(tt.args["shortener_server_address"], tt.args["shortener_base_url"], "")
-			rw, r := sendRequest(http.MethodPost, URL, strings.NewReader(tt.body))
-
-			repository := storage.NewMemoryShortenerRepository(args.BaseURL)
-			linkShortener := shortener.New(repository)
-
-			handler.New(linkShortener).Post(rw, r)
+			config := &Config{Port: "8080"}
+			args := NewArgs(config)
 
-			res := rw.Result()
-			defer responseClose(t, res)
+			args.ParseFlags(tt.args)
 
-			body, err := io.ReadAll(res.Body)
-
-			if err != nil {
-				t.Fatalf("could not read response:%v", err)
+			if args.ServerAddress != tt.expected.ServerAddress {
+				t.Errorf("expected ServerAddress to be '%s', got '%s'", tt.expected.ServerAddress, args.ServerAddress)
 			}
-
-			if string(body) != tt.wantError {
-				t.Errorf("error handler not correct: got %v want %v", string(body), tt.wantError)
+			if args.BaseURL != tt.expected.BaseURL {
+				t.Errorf("expected BaseURL to be '%s', got '%s'", tt.expected.BaseURL, args.BaseURL)
 			}
-
-			if res.StatusCode != tt.wantStatusCode {
-				t.Errorf("excpected status: got %v want %v", res.StatusCode, tt.wantStatusCode)
+			if args.FileStoragePath != tt.expected.FileStoragePath {
+				t.Errorf("expected FileStoragePath to be '%s', got '%s'", tt.expected.FileStoragePath, args.FileStoragePath)
+			}
+			if args.DatabaseDNS != tt.expected.DatabaseDNS {
+				t.Errorf("expected DatabaseDNS to be '%s', got '%s'", tt.expected.DatabaseDNS, args.DatabaseDNS)
```

Comment:
Вместо сравнения полей по оддельности удобно использовать `cpm.Diff(...)` для сравнения структур:
```suggestion
	if diff := cmp.Diff(tt.expected, args); diff != "" {
		t.Fatalf("unexpected config parameters (-want +got):\n%s", diff)
	}
```